app/internal/security/ouath2.py

```
@@ -30,19 +39,38 @@ def verify_password(plain_password: str, hashed_password: str) -> bool:
     return pwd_context.verify(plain_password, hashed_password)
 
 
+async def authenticate_user_by_email(
+        db: Session, user: schema.ForgotPassword,
+        email: bool = False) -> Union[schema.ForgotPassword, bool]:
+    """
+    Verifying database record by username.
+    Comparing given email to database record,
+    """
+    db_user = await User.get_by_username(db=db, username=user.username)
+    if not db_user:
+        return False
```

Prefer to raise an exception

---

raise an exception to the user? what i'm doing here is re-rendering the form, asking him to check his credentials